src/run.rs

```diff
@@ -276,8 +277,49 @@ pub fn clean_up(
     worktree_branch: Option<&str>,
     project_name: Option<&str>,
     subprocesses: &mut [Child],
+    prompt: bool,
 ) -> Result<()> {
-    if !delete_file.exists() {
+    let delete = if !delete_file.exists() {
+        false
+    } else if prompt {
+        println!("Are you sure you want to delete this project? (Y/n)");
+        loop {
+            let mut input = String::new();
```

Comment:
To avoid re-allocating you can move this out of the loop and clear the buffer.

(I know the doc says "You do not need to clear the buffer before appending." but they also clearly say that the bytes are appended to the string and they clear the buffer in the example :woman_shrugging: so I'm not sure why they say that tbh)

What happens when you use the `clear()` method is that the buffer is not deallocated. https://doc.rust-lang.org/std/string/struct.String.html#method.clear Only the length is set back to 0 which would work as an empty string.